File: common/src/main/java/mconi/common/sim/model/PressureBand.java
package mconi.common.sim.model;

public enum PressureBand
{
	VACUUM,
	THIN,
	BREATHABLE,
	OVERPRESSURE;

	public static PressureBand fromKpa(double pressureKpa)
	{
		if (pressureKpa < 20.0D) { return VACUUM; }
		if (pressureKpa < 70.0D) { return THIN; }
		if (pressureKpa <= 180.0D) { return BREATHABLE; }
		return OVERPRESSURE;
	}
}

File: common/src/main/kotlin/mconi/common/sim/model/PressureBand.kt
package mconi.common.sim.model

enum class PressureBand {
    VACUUM,
    THIN,
    BREATHABLE,
    OVERPRESSURE;

    companion object {
        @JvmStatic
        fun fromKpa(pressureKpa: Double): PressureBand {
            if (pressureKpa < 20.0) {
                return VACUUM
            }
            if (pressureKpa < 70.0) {
                return THIN
            }
            if (pressureKpa <= 180.0) {
                return BREATHABLE
            }
            return OVERPRESSURE
        }
    }
}
